optimum/commands/export/openvino.py

```diff
@@ -361,7 +361,7 @@ def run(self):
             model.save_pretrained(self.args.output)
             if not self.args.disable_convert_tokenizer:
                 maybe_convert_tokenizers(library_name, self.args.output, model, task=task)
-        elif (task.startswith("text-generation") or task == "image-text-to-text") and quantize_with_dataset:
+        elif task.startswith("text-generation") and quantize_with_dataset or task == "image-text-to-text":
```

Comment:
Right, the fix for the memory consumption for Vision models is WIP in OpenVINO.

Reply:
If memory use is 33 GB there is no way to export the model on AI PC. With the existing method, specifying group size 16, this model can be exported on client with 32GB RAM. Can we keep that functionality? 

I only tested with one image so far, but output looked fine, which is pretty surprising to me given that all the models were unexpectedly INT4.

Reply:
I believe in theory it is possible, but will require some code refactoring. But I would prefer to address the increased memory during the current way of conversion because it also affects export via Python API.

Reply:
I've made some findings regarding the peak memory differences. They arise from different weights precision during model conversion. When converting PT vision-language model from python API **or** with optimum-cli without compression (`--weight-format fp32`), models are exported in fp32. When model conversion is run via optimum-cli with `--weight-format in8/int4` (on the main branch) and original models weights are in bf16, then models are converted with bf16 patching (see #778). Exporting with bf16 patching lowers peak memory compared to fp32 case so that is why we see differences here.

In this PR, conversion of VLMs with compression via optimum-cli is made to follow the python API conversion path. That is why models start to be exported in fp32 instead of bf16 and we see a peak memory difference. 

I've added the following fix at modeling_visual_language.py to address this issue:
```python
ov_config = OVConfig(dtype="fp32" if load_in_8bit is False else "auto")
```

With this, VLM is first exported to bf16 and then compressed. I now observe 23 GB peak memory for the following command
```
optimum-cli export openvino --model openbmb/MiniCPM-V-2_6 --trust-remote-code --weight-format int4 ./tmp
```